src/linter/dotLibrary/DotLibraryLinter.ts

```diff
@@ -0,0 +1,99 @@
+import {LintMessageSeverity} from "../LinterContext.js";
+import LinterContext from "../LinterContext.js";
+import {deprecatedLibraries} from "../../utils/deprecations.js";
+import {SaxEventType, Tag as SaxTag} from "sax-wasm";
+import {parseXML} from "../../utils/xmlParser.js";
+import {ReadStream} from "node:fs";
+import {RULES, MESSAGES, formatMessage} from "../linterReporting.js";
+
+export default class DotLibraryLinter {
+	#contentStream;
+	#resourcePath;
+	#context: LinterContext;
+
+	constructor(resourcePath: string, contentStream: ReadStream, context: LinterContext) {
+		this.#contentStream = contentStream;
+		this.#resourcePath = resourcePath;
+		this.#context = context;
+	}
+
+	async lint() {
+		try {
+			const dotLibraryDependencyTags = await this.#parseDotLibrary(this.#contentStream);
+			this.#analyzeDeprecatedLibs(dotLibraryDependencyTags);
+		} catch (err) {
+			const message = err instanceof Error ? err.message : String(err);
+			this.#context.addLintingMessage(this.#resourcePath, {
+				severity: LintMessageSeverity.Error,
+				message,
+				ruleId: RULES["ui5-linter-parsing-error"],
+				fatal: true,
+			});
+		}
+	}
+
+	async #parseDotLibrary(contentStream: ReadStream): Promise<SaxTag[]> {
+		const libs = new Set();
+		const tagsStack: string[] = [];
+		const libNamePath = ["library", "dependencies", "dependency"];
+		await parseXML(contentStream, (event, tag) => {
+			if (!(tag instanceof SaxTag)) {
+				return;
+			}
+
+			if (event === SaxEventType.OpenTag && !tag.selfClosing) {
+				tagsStack.push(tag.value);
+			} else if (event === SaxEventType.CloseTag && !tag.selfClosing) {
+				tagsStack.pop();
+			}
+
+			if (event === SaxEventType.CloseTag &&
+				tag.value === "libraryName") {
+				const path = tagsStack.slice(-1 * libNamePath.length);
```

Comment:
When changing this to
```js
const path = tagsStack;
```

all tests still succeed. In addition, the path is only matched when there are exactly 3 tags in the stack. To me this matches with the expected behavior.

Otherwise (with the slice), the path would still be matched when there are more tags at root level (e.g. `foo > library > dependencies > dependency`).

Reply:
Done